packages/fuels-macros/src/parse_utils.rs

```diff
@@ -80,3 +81,118 @@ where
         .map(|duplicates| generate_duplicate_error(&duplicates))
         .validate_no_errors()
 }
+
+pub fn validate_and_extract_generic_types(generics: &Generics) -> syn::Result<Vec<&TypeParam>> {
+    generics
+        .params
+        .iter()
+        .map(|generic_param| match generic_param {
+            GenericParam::Type(generic_type) => Ok(generic_type),
+            GenericParam::Lifetime(lifetime) => {
+                Err(Error::new_spanned(lifetime, "Lifetimes not supported"))
+            }
+            GenericParam::Const(const_generic) => Err(Error::new_spanned(
+                const_generic,
+                "Const generics not supported",
+            )),
+        })
+        .collect()
+}
+
+pub(crate) struct Members {
+    names: Vec<Ident>,
+    types: Vec<TokenStream>,
+    fuels_types_path: TokenStream,
+}
+
+impl Members {
+    pub(crate) fn names(&self) -> impl Iterator<Item = &Ident> + '_ {
+        self.names.iter()
+    }
+
+    pub(crate) fn names_as_strings(&self) -> impl Iterator<Item = TokenStream> + '_ {
+        self.names().map(|ident| {
+            let name = ident.to_string();
+            quote! {#name.to_string()}
+        })
+    }
+    pub(crate) fn param_type_calls(&self) -> impl Iterator<Item = TokenStream> + '_ {
+        let fuels_types_path = self.fuels_types_path.to_token_stream();
+        self.types.iter().map(move |ty| {
+            quote! { <#ty as #fuels_types_path::traits::Parameterize>::param_type() }
+        })
+    }
+}
+
+pub(crate) fn extract_struct_members(
+    fields: DataStruct,
+    fuels_types_path: TokenStream,
+) -> syn::Result<Members> {
+    let named_fields = match fields.fields {
+        Fields::Named(named_fields) => Ok(named_fields.named),
+        Fields::Unnamed(fields) => Err(Error::new_spanned(
+            fields.unnamed,
+            "Tuple-like structs not supported",
+        )),
+        _ => {
+            panic!("This cannot happen in valid Rust code. Fields::Unit only appears in enums")
+        }
+    }?;
+
+    let (names, types) = named_fields
+        .into_iter()
+        .map(|field| {
+            let name = field
+                .ident
+                .expect("FieldsNamed to only contain named fields!.");
```

Comment:
```suggestion
                .expect("FieldsNamed to only contain named fields.");
```

Sorry, I wish my eyes couldn't catch such small typos 😂